src/org/nejrasm/zadaca4oop/task1/Duck.java

```diff
@@ -0,0 +1,13 @@
+package org.nejrasm.zadaca4oop.task1;
+
+public class Duck implements Flight, Swim {
+    @Override
+    public String fly() {
+        return new Pegasus().fly();
```

Comment:
:D this does go a bit too far. Separation of concern is preferred. Also you had to create a new Pegasus object just to reuse it's method. What if Pegasus actually had arguments for it's constructor, how would you provide them?

One way to do what you wanted was following:
1.
```
public class FlyWithWIngs implements Flight {
    public String fly() {
        return "Flies with wings";
    }
}
```
Give Duck and pegasus a flying behaviour:
```
public class Duck implements Flight {
    Flight flyingBehaviour;
    public Duck() {
        flyingBehaviour = new FlyWithWings();
    }
    public String fly() {
        flyingBehaviour.fly();
    }
}
```

In this particular case it is not even required for duck or pegasus to implement flight. Anyway this was not the goal of the exercise.

I appreciate that you try to reuse code, that is very good, but just make the same method for simplicity now.

Reply:
Noted :)